### PR TITLE
[MODINREACH-97] - Report Unshipped Item Received to Owning Site for Item Hold

### PR DESCRIPTION
**Pull request by aleksandr2110**
## Purpose

Add an endpoint for report unshipped item received to owning site for item hold
https://issues.folio.org/browse/MODINREACH-97

## Approach
 

InnReachCirculationController modified.
InnReachCirculationService modified.
OpenAPI configuration modified.
InnReachCirculationController tests added.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java [hunk 1]
```diff
@@ -163,6 +164,23 @@ public InnReachResponseDTO cancelItemHold(String trackingId, String centralCode,
...
+    var transaction = getTransaction(trackingId, centralCode);
+
+    if (transaction.getState() == InnReachTransaction.TransactionState.ITEM_SHIPPED) {
+      throw new IllegalArgumentException("Unexpected transaction state: " + transaction.getState());
+    }
+
+    if (transaction.getState() == InnReachTransaction.TransactionState.ITEM_HOLD) {
+      transaction.setState(RECEIVE_UNANNOUNCED);
+    }
+    transactionRepository.save(transaction);
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java]: lets move `transactionRepository.save(transaction);` to the ITEM_HOLD condition block above, as it's the only place where transaction data is actually being updated

**Review by oliinyko — APPROVED** [review]

File: src/main/java/org/folio/innreach/controller/d2ir/InnReachCirculationController.java [hunk 2]
```diff
@@ -72,4 +72,13 @@
     return ResponseEntity.ok(innReachResponse);
   }
 
+  @Override
+  @PutMapping("/unshipedtemhold/{trackingId}/{centralCode}")
```
Comment (dmtkachenko) [inline src/main/java/org/folio/innreach/controller/d2ir/InnReachCirculationController.java]: URL is `receiveunshipped/<trackingId>/<centralCode>`
not `unshipedtemhold...`

File: src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java [hunk 3]
```diff
@@ -231,4 +234,61 @@ void processCancelItemHoldRequest_whenItemIsCheckedOut() {
...
+  void precessReportUnshippedItemReceived_whenTransactionItemHold() {
+    var transaction = repository.findByTrackingIdAndCentralServerCode(PRE_POPULATED_TRACKING_ID,
+      PRE_POPULATED_CENTRAL_CODE).get();
+    transaction.setState(ITEM_HOLD);
+    repository.save(transaction);
+
+    var transactionHoldDTO = createTransactionHoldDTO();
+
+    var responseEntity = testRestTemplate.exchange(
+      "/inn-reach/d2ir/circ/unshipedtemhold/{trackingId}/{centralCode}", HttpMethod.PUT,
```
Comment (dmtkachenko) [inline src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java]: `receiveunshipped`

File: src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java (hunk 1 above)
Comment (aleksandr2110) [reply src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java]: Done

**Review by dmtkachenko — APPROVED** [review]

File: src/main/java/org/folio/innreach/controller/d2ir/InnReachCirculationController.java (hunk 2 above)
Comment (aleksandr2110) [reply src/main/java/org/folio/innreach/controller/d2ir/InnReachCirculationController.java]: Done

File: src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java (hunk 3 above)
Comment (aleksandr2110) [reply src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java]: Done

**Pull request merged**